api/views.py

```diff
@@ -99,4 +99,284 @@ def redirect_login(req):
     user = User.objects.get(username=req.user.username)
     acc = SocialAccount.objects.get(user=user)
     token = SocialToken.objects.get(account=acc)
-    return HttpResponseRedirect('/login?status=passed&token=' + token.token + '&username=' + user.username + '&userid=' + str(user.id))
+    if not str(user.id) == acc.uid:
+        tmp = user
+        user = user.delete()
+        tmp.id = acc.uid
+        tmp.save()
+        acc.user = tmp
+        acc.save()
+        return HttpResponseRedirect('/login?status=passed&token=' + token.token + '&username=' + tmp.username + '&user_id=' + str(tmp.id))
```

Comment:
Instead of hardcoding the URL like `/login`, can we use the name of the URL here using `reverse()`?